src/FileDownloader.ts

```diff
@@ -84,16 +85,23 @@ export default class FileDownloader implements IFileDownloader {
                 onDownloadProgressChange
             );
 
-            const writeStream: Writable = settings?.shouldUnzip ?? false
-                ? ZipDecompressor(tempFileDownloadPath)
-                : fs.createWriteStream(tempFileDownloadPath);
-
-            // Start the download and wait for it to completely finish writing to disk
-            // The first element in the pipeline must be the readableStream and the last element must be the writableStream
+            const writeStream = fs.createWriteStream(tempZipFileDownloadPath);
             const pipelinePromise = pipelineAsync([downloadStream, writeStream]);
             const writeStreamClosePromise = new Promise(resolve => writeStream.on(`close`, resolve));
             await Promise.all([pipelinePromise, writeStreamClosePromise]);
 
+            const unzipDownloadedFileAsyncFn = async (): Promise<void> => {
+                await fs.promises.access(tempZipFileDownloadPath);
+                if (settings?.shouldUnzip ?? false) {
+                    await extractZip(tempZipFileDownloadPath, { dir: tempFileDownloadPath });
+                }
+                else {
+                    await fs.promises.copyFile(tempZipFileDownloadPath, tempFileDownloadPath);
```

Comment:
It's looks strange to download in a ZIP location (we're not even a ZIP), then to copy in another tempFileDownloadPath, and then only to rename to the final name. Could we directly download in `tempFileDownloadPath` to reduce copies and the risk to have any issues when we copy/delete/rename?

Reply:
Sure, I can simplify the locations. 
But I didn't get when you say we are not even zip?

Reply:
What I meant it that here, we're also handling the case where `settings?.shouldUnzip` == false. At this point, we're trying to download something that isn't ZIP related, so it's strange to download it under a ZIP name, and enter the unzipDownloadedFileAsyncFn function.